python/mxnet/ndarray/ndarray.py

```diff
@@ -1048,7 +1048,10 @@ def _advanced_index_to_array(idx, ax_len, ctx):
 
         The ``ax_len`` is used to convert `slice` objects to integer arrays.
         """
-        idx_dtype = 'int32'
+        if sys.version_info[0] > 2 and _int64_enabled():
+            idx_dtype = 'int64'
+        else:
+            idx_dtype = 'int32'
```

Comment:
Performing this check since idx can be an instance of NDArray, list, tuple, slice(python), integer_type etc. Also, earlier idx was hard coded to be 'int32' therefore with `Large Tensor enabled` it makes sense to use only 'int64'.

Reply:
Have we measured runtime difference by adding this check in ndarray.py? Can we some simple test?

Reply:
ran test_ndarray.py on my branch and on master. Both take 20-21 secs. Master is taking 1-2 secs more than my change.